migration/migrator_range.go

```diff
@@ -0,0 +1,122 @@
+package migration
+
+import (
+	"bytes"
+	"context"
+	"math/big"
+	"runtime"
+
+	"github.com/holiman/uint256"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+var (
+	NumProcessAccount = runtime.NumCPU()
+	NumProcessStorage = runtime.NumCPU()
+)
+
+var hashSpace = new(big.Int).Exp(common.Big2, common.Big256, nil)
+
+// hashRange is a utility to handle ranges of hashes, Split up the
+// hash-space into sections, and 'walk' over the sections
+type hashRange struct {
+	current *uint256.Int
+	step    *uint256.Int
+}
+
+// newHashRange creates a new hashRange, initiated at the start position,
+// and with the step set to fill the desired 'num' chunks
+func newHashRange(start common.Hash, num uint64) *hashRange {
```

Comment:
It may not matter, but wouldn't it be num + 1 chunks instead of num chunks?

Reply:
This comment comes from the `eth/protocols/snap/range.go` file.